syft/pkg/cataloger/golang/scan_bin.go

```diff
@@ -25,17 +25,27 @@ func scanFile(reader unionReader, filename string) ([]*debug.BuildInfo, []string
 	// with more than one binary
 	readers, err := getReaders(reader)
 	if err != nil {
-		log.Warnf("golang cataloger: opening binary: %v", err)
+		log.Warnf("golang cataloger: failed to open a binary: %v", err)
 		return nil, nil
 	}
 
 	var builds []*debug.BuildInfo
 	for _, r := range readers {
 		bi, err := buildinfo.Read(r)
+
+		// note: the stdlib does not export the error we need to check for
 		if err != nil {
-			log.Warnf("golang cataloger: scanning file %s: %v", filename, err)
+			if err.Error() == "not a Go executable" {
+				// since the cataloger can only select executables and not distinguish if they are a go-compiled
+				// binary, we should not show warnings/logs in this case.
+				return nil, nil
+			}
+			// in this case we could not read the or parse the file, but not explicitly because it is not a
```

Comment:
>but not explicitly because it is not a go-compiled binary (though it still might be)

nit: this phrasing is just a little difficult to parse, I think because of having "but not ... not ... though ...". If we can think of a more direct way to say this, I think that'd help (still thinking 🤔 )

Reply:
Maybe something like "We don't know if this is a Go binary, but we weren't able to read it, so we should alert the user."?

(I'm still making sure I understand the intent of this comment, so let me know if I'm off-base!)